perception/autoware_camera_streampetr/lib/network/camera_data_store.cpp

```diff
@@ -94,17 +93,31 @@ CameraDataStore::CameraDataStore(
     cudaStreamCreate(&streams_[i]);
   }
 
+  // Initialize undistortion map storage
+  undistort_map_x_gpu_.resize(rois_number, nullptr);
+  undistort_map_y_gpu_.resize(rois_number, nullptr);
+  undistortion_maps_computed_.resize(rois_number, false);
+
   is_frozen_ = false;
   active_updates_ = 0;
+}
 
-  // Validate downsample_factor when distorted image is used
-  if (is_distorted_image_) {
-    const bool invalid_downsample_factor = (downsample_factor_ <= 0.0 || downsample_factor_ > 1.0);
-    if (invalid_downsample_factor) {
-      throw std::runtime_error(
-        "downsample_factor must be in range (0,1] when is_distorted_image is true");
+CameraDataStore::~CameraDataStore()
+{
+  // Clean up GPU memory for undistortion maps
+  for (size_t i = 0; i < rois_number_; ++i) {
+    if (undistort_map_x_gpu_[i] != nullptr) {
+      cudaFree(undistort_map_x_gpu_[i]);
+    }
+    if (undistort_map_y_gpu_[i] != nullptr) {
+      cudaFree(undistort_map_y_gpu_[i]);
     }
   }
+
+  // Clean up CUDA streams
+  for (auto & stream : streams_) {
+    cudaStreamDestroy(stream);
+  }
 }
 
 void CameraDataStore::update_camera_image(
```

Comment:
[bug fix]
after L144, 
```cpp
if (is_distorted_image_ && camera_info_list_[camera_id]) {
    image_input_tensor = process_distorted_image(camera_id, input_camera_image_msg, params);
} else {
    image_input_tensor = process_regular_image(input_camera_image_msg, params, camera_id);
}
```

image_input_tensor could be nullptr but there are no escape. Just add nullptr escape to prevent invalid access.